### ISSUE TITLE
Application.Current will be null

### ISSUE THREAD

**Issue by ShrlAlgo** [lepoco/wpfui#662]
### Describe the bug

![image](https://github.com/lepoco/wpfui/assets/40654941/224989c3-2d97-417d-834e-ae6e9b4795b8)

### To Reproduce

Revit Plugins Development

### Expected behavior

In case of software plug-in development，this Application.Current will be null

### OS version

Windows 11

### .NET version

net48

### WPF-UI NuGet version

3.0.0 preview.3

**Comment by 23W** [comment]
The same bug in release 2.0.3.
We need a documentation how to avoid this issues https://github.com/lepoco/wpfui/issues/668

**Comment by chucker** [comment]
Yep. I've had to copy some of WPFUI's code to avoid this issue, as I can't port my entire app in one go.

This is particularly an issue (for me) with calls hardcoded to `Application.Current.Resources`.

Perhaps it would be better if you could have an `IWpfUiResourceService` that you pass to those methods, with a default implementation that points to `Application.Current.Resources`, but a way for you to supply your own.

**Comment by 23W** [comment]
> Perhaps it would be better if you could have an `IWpfUiResourceService`
And this `IWpfUiResourceService` should be public to be able to change theme colors at runtime.
